app/models/concerns/orchestration/dhcp.rb

```diff
@@ -135,21 +135,22 @@ def dhcp_record_name(record_mac)
   end
 
   def queue_dhcp
+    logger.warn("Provisioning interface #{self} has no name set, cannot queue DHCP update") if provision? && hostname.blank?
```

Comment:
During testing I figured out that NIC must have name set in order to orchestrate DHCP. Foreman does not warn if this is not present, DHCP is just silently skipped. This should help debugging this.